atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -125,12 +124,18 @@ private long extractUpperLimit(ColumnOrSuperColumn result) {
                 }
             }
         });
+
+        DebugLogger.logger.info("[GET] Setting cached timestamp limit to {}.", currentLimit);
+        currentLimit = upperLimit;
+        return currentLimit;
```

Comment:
In general, we bung too much stuff inside `CassandraClientPool.runWithRetry` - the only things in there should be talking to the database.